tests/integration/steps/stargate_steps.go

```diff
@@ -0,0 +1,133 @@
+	g(t).Expect(err).To(BeNil(), "Failed connecting to Stargate")
+	stargateResponse := response.Body()
+	var genericJson map[string]interface{}
+	json.Unmarshal(stargateResponse, &genericJson)
+	return genericJson["authToken"].(string)
+}
+
+func CreateStargateDocumentNamespace(t *testing.T, token string) string {
+	docNamespace := fmt.Sprintf("stargate%s", time.Now().Format("2006010215040507"))
+	docNamespaceJson := fmt.Sprintf("{\"name\":\"%s\"}", docNamespace)
+	restClient := resty.New()
+	response, err := restClient.R().
+		SetHeader("Content-Type", "application/json").
+		SetHeader("X-Cassandra-Token", token).
+		SetBody(docNamespaceJson).
+		Post("http://stargate.127.0.0.1.nip.io:8082/v2/schemas/namespaces")
+	g(t).Expect(err).To(BeNil(), "Failed creating Stargate document namespace")
+	stargateResponse := string(response.Body())
+	g(t).Expect(stargateResponse).To(Equal(docNamespaceJson), fmt.Sprintf("Unexpected response from Stargate: %s", stargateResponse))
+	return docNamespace
+}
+
+const (
+	awesomeMovieDirector = "Zack Snyder"
+	awesomeMovieName     = "Watchmen"
+)
+
+func WriteStargateDocument(t *testing.T, token, docNamespace string) string {
+	awesomeMovieDocument := fmt.Sprintf("{\"Director\":\"%s\",\"Name\":\"%s\"}", awesomeMovieDirector, awesomeMovieName)
+	documentId := fmt.Sprintf("watchmen%s", time.Now().Format("2006010215040507"))
+	restClient := resty.New()
+	response, err := restClient.R().
+		SetHeader("Content-Type", "application/json").
+		SetHeader("X-Cassandra-Token", token).
+		SetBody(awesomeMovieDocument).
+		Put(fmt.Sprintf("http://stargate.127.0.0.1.nip.io:8082/v2/namespaces/%s/collections/movies/%s", docNamespace, documentId))
+	g(t).Expect(err).To(BeNil(), "Failed creating Stargate document")
+	stargateResponse := string(response.Body())
+	expectedResponse := fmt.Sprintf("{\"documentId\":\"%s\"}", documentId)
+	g(t).Expect(stargateResponse).To(Equal(expectedResponse))
+	return documentId
+}
+
+func CheckStargateDocumentExists(t *testing.T, token, docNamespace, documentId string) {
+	document := readStargateDocument(t, token, docNamespace, documentId)
+	var genericJson map[string]interface{}
+	json.Unmarshal(document, &genericJson)
+	g(t).Expect(genericJson["documentId"].(string)).To(Equal(documentId))
+	g(t).Expect(genericJson["data"].(map[string]interface{})["Director"].(string)).To(Equal(awesomeMovieDirector))
+	g(t).Expect(genericJson["data"].(map[string]interface{})["Name"].(string)).To(Equal(awesomeMovieName))
+}
+
+func readStargateDocument(t *testing.T, token, docNamespace, documentId string) []byte {
+	restClient := resty.New()
+	response, err := restClient.R().
+		SetHeader("Content-Type", "application/json").
+		SetHeader("X-Cassandra-Token", token).
+		Get(fmt.Sprintf("http://stargate.127.0.0.1.nip.io:8082/v2/namespaces/%s/collections/movies/%s", docNamespace, documentId))
+	g(t).Expect(err).To(BeNil(), "Failed reading Stargate document")
+	return response.Body()
```

Comment:
Same possible issue with response being nil, throwing an exception here.

Reply:
the call to `Expect(err)...` on the previous line protects us here. If response is nil indeed, then something went totally wrong in the rest call and we shouldn't reach the `return` instruction. 